libs/gpu/src/dawn.zig

```diff
@@ -57,3 +59,19 @@ pub const TogglesDeviceDescriptor = extern struct {
         };
     }
 };
+
+const AdapterPropertiesPowerPreference = extern struct {
+    chain: ChainedStructOut = .{
+        .next = null,
+        .s_type = .dawn_adapter_properties_power_preference,
+    },
+    power_preference: PowerPreference = .undefined,
+};
+
+const BufferDescriptorErrorInfoFromWireClient = extern struct {
+    chain: ChainedStruct = .{
+        .next = null,
+        .s_type = .dawn_buffer_descriptor_error_info_from_wire_client,
+    },
+    out_of_memory: bool = false,
```

Comment:
We can see in [dawn.json `"chain roots": ["buffer descriptor"],`](https://github.com/hexops/dawn/blob/generated-2023-01-12.1673546526/dawn.json#L3022-L3025) which tells us this is a valid type to specify in `gpu.Buffer.Descriptor.next_in_chain`, so we need to update `gpu.Buffer.Descriptor.next_in_chain` data type to be an `extern union` where you can specify this type, similar to what we do for `Device.Descriptor.next_in_chain`:

https://github.com/hexops/mach/blob/c02578721a2aa695a7c80802ab8903d4c3f75e41/libs/gpu/src/device.zig#L46-L52